src/logging.h

```diff
@@ -178,7 +178,7 @@ static inline void LogPrintf(const char* fmt, const Args&... args)
 // evaluating arguments when logging for the category is not enabled.
 #define LogPrint(category, ...)              \
     do {                                     \
-        if (LogAcceptCategory((category))) { \
+        if ((category) == BCLog::LogFlags::ALL || LogAcceptCategory((category))) { \
```

Comment:
Why is this change needed?

Reply:
Added this to ensure the FlushStateToDisk call would be logged even if no categories were active, which I believe is consistent with the current behavior - though I am not certain it is necessary.

Reply:
I am not convinced at the moment that this is necessary. `Timer::Log` has the following implementation:
```
    void Log(const std::string& msg)
    {
        const std::string full_msg = this->LogMsg(msg);

        if (m_log_category == BCLog::LogFlags::ALL) {
            LogPrintf("%s\n", full_msg);
        } else {
            LogPrint(m_log_category, "%s\n", full_msg);
        }
    }
```
To me, this implies that `LogPrint(BCLog::LogFlags::ALL…` is never called. And it shouldn't, it's not valid usage.

Reply:
I've removed this change.